canonical/websites-content-system
# Create new copydoc file if webpage is new

## Done

 - Created a new copydoc when a webpage is requested

## QA

 - Check that the demo runs at https://websites-content-system-60.demos.haus/
 - Make a new page request at `/request-changes`, similar to the below
```
           {
                "due_date": "2022-01-01",
                "reporter_id": 1,
                "webpage_id": 31,
                "type": 2,
                "description": "This is a description",
            }
```
 - A new copydoc should be created at [testautomation](https://drive.google.com/drive/folders/1KzfeHcGqQtRxL8CUxMB9uQk5FHX9Jdir?ths=true) with the file nested according to the url e.g. `canonical.com/careers/results` 
   - The copydoc should have the right name
   - The copydoc should be in the correct folder i.e the folder should have the project and the webpage name e.g ubuntu.com/careers

## Note
- There is a short delay, though sometimes up to 5 mins before the doc is visible in the UI after a successful request
- Google drive sometimes fails to create a document, and returns a 500 error

## Fixes

 - Fixes [WD-13381](https://warthogs.atlassian.net/browse/WD-13381?atlOrigin=eyJpIjoiNWU4ZDhkM2Y1OWI4NDM5OGIzNGY4ZGVmMDY5MzI2ZjkiLCJwIjoiaiJ9)


[WD-13381]: https://warthogs.atlassian.net/browse/WD-13381?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
